client/header/activity-panel/index.js

```diff
@@ -71,30 +68,35 @@ export class ActivityPanel extends Component {
 		};
 	}
 
-	togglePanel( tabName ) {
-		const { isPanelOpen, currentTab } = this.state;
-
-		// If a panel is being opened, or if an existing panel is already open and a different one is being opened, record a track.
-		if ( ! isPanelOpen || tabName !== currentTab ) {
-			recordEvent( 'activity_panel_open', { tab: tabName } );
-		}
-
+	togglePanel( { name: tabName }, isTabOpen ) {
 		this.setState( ( state ) => {
-			if ( tabName === state.currentTab || state.currentTab === '' ) {
-				return {
-					isPanelOpen: ! state.isPanelOpen,
-					currentTab: tabName,
-					mobileOpen: ! state.isPanelOpen,
-				};
-			}
-			return { currentTab: tabName, isPanelSwitching: true };
+			const isPanelSwitching =
+				tabName !== state.currentTab &&
+				state.currentTab !== '' &&
+				isTabOpen &&
+				state.isPanelOpen;
+
+			return {
+				isPanelOpen: isTabOpen,
+				mobileOpen: isTabOpen,
+				currentTab: tabName,
+				isPanelSwitching,
+			};
 		} );
 	}
 
+	closePanel() {
+		this.setState( ( state ) => ( {
+			...state,
```

Comment:
Doesn't React automatically merge the state when calling `setState()`?  Is there an advantage to destructuring the state here?

Reply:
@joshuatf Thats a good point, it does merge state on the first level (shallow only), I tend to do this out of habit because it doesn't merge state if you're dealing with nested objects. I can remove this though.